### PR TITLE
test(modules): Add unit tests for CommModule communication panel #4923

### LINKED ISSUE #4923: [Bug]: Add unit tests for CommModule communication panel handler (0% coverage)
### Description

`esp.program.modules.handlers.commmodule.py` has **0% test coverage** and no test file exists in `tests/`. This handler powers the **Communication Panel** — the mass email system that admins use to send emails to hundreds of students, teachers, and volunteers at once.

The CommModule handles:
- Rendering the email composition interface with recipient filters
- Processing and sending mass emails to filtered user lists
- Email template management
- Recipient list generation based on program roles and registration status

Untested email-blast logic is extremely high-risk: a silent failure here means hundreds of students could miss critical program updates (schedule changes, room reassignments, cancellations). An incorrect recipient filter could send sensitive admin-only content to the wrong audience.

### Proposed Tests

1. **Main view rendering:** Verify the comm panel page loads correctly with proper recipient filter options and email composition form.
2. **Recipient filtering:** Test that user lists are correctly generated based on program roles (students, teachers, admins) and registration status filters.
3. **Email sending logic:** Mock `send_mail` and verify correct recipients, subject, and body are passed through.
4. **Permission checks:** Verify that only authorized admins can access the comm panel and send emails.
5. **Edge cases:** Empty recipient lists, malformed email content, missing template fields.

### Related Issue
Parent: #3780

### PR DESCRIPTION
**Pull request by moksha-hub**
## Summary
Add 12 new tests for previously untested CommModule methods, complementing the existing CommunicationsPanelTest.

## Tests Added
| Test Class | Tests | Coverage |
|---|---|---|
| ProgramUrlsInTextTest | 4 | _program_urls_in_text helper |
| ApproxNumRecipientsTest | 1 | approx_num_of_recipients |
| CommPanelViewsTest | 7 | views + admin permissions |

Existing CommunicationsPanelTest is untouched.

## Related Issue
Fixes #4923 | Parent: #3780

### CODE REVIEW

**Comment by moksha-hub** [comment]
@Oval17 have a review.

File: esp/esp/program/modules/tests/commpanel.py [hunk 1]
```diff
@@ -170,3 +170,228 @@ def test_program_date_variables_in_comms(self):
...
+            'filterid': filterid,
+            'listcount': '3',
+            'from': 'info@testserver.learningu.org',
+            'replyto': 'replyto@testserver.learningu.org',
+            'subject': 'Test Subject',
+            'body': 'Test Body',
+        }
+        response = self.client.post(
+            '/manage/%s/maincomm2' % self.program.getUrlBase(), post_data)
+        self.assertEqual(response.status_code, 200)
```
Comment (Oval17) [inline esp/esp/program/modules/tests/commpanel.py:354]: This only asserts status_code == 200 , a 200 error page would pass too. Can we add assertTemplateUsed for the step2 template so the test actually confirms the right page rendered?

File: esp/esp/program/modules/tests/commpanel.py [hunk 2]
```diff
@@ -170,3 +170,228 @@ def test_program_date_variables_in_comms(self):
...
+            'filterid': filterid,
+            'listcount': listcount,
+            'subject': 'Preview Subject',
+            'body': '<p>Preview Body</p>',
+            'from': 'info@testserver.learningu.org',
+            'replyto': 'replyto@testserver.learningu.org',
+        }
+        response = self.client.post(
+            '/manage/%s/commprev' % self.program.getUrlBase(), post_data)
+        self.assertEqual(response.status_code, 200)
```
Comment (Oval17) [inline esp/esp/program/modules/tests/commpanel.py:404]: same as above!

**Review by Oval17 — COMMENTED** [review]
Most of comments from github copilot is addressed, There are few `import re as re_mod` , though re is imported at module level.These are just extra lines but can be ignored
Rest Looks good to me!!
Thanks

**Comment by Oval17** [comment]
LGTM.
